internal/driver/fetch.go

```diff
@@ -401,34 +401,42 @@ mapping:
 		if m.File != "" {
 			baseName = filepath.Base(m.File)
 		}
-
-		for _, path := range filepath.SplitList(searchPath) {
-			var fileNames []string
-			if m.BuildID != "" {
-				fileNames = []string{filepath.Join(path, m.BuildID, baseName)}
+		var fileNames []string
+		if m.BuildID != "" {
+			// gdb-style debuginfo format puts debug symbols into paths like .build-id/01/23456789abcdef.debug
+			debugInfoName := filepath.Join(".build-id", m.BuildID[:2], m.BuildID[2:]+".debug")
```

Comment:
This should probably check len(m.BuildID)?

Reply:
Good point.